docs/gaps/README.md

```diff
@@ -0,0 +1,190 @@
+first iteration of a GAP.
+1. **References:** A list of references that include links to discussions
+pertaining to the GAP as well as any external links relevant to the proposed
+changes.
+
+A GAP document may also include the following optional sections.
+
+1. **Acknowledgements:** Any relevant acknowledgements to people (who are not
+sponsors or contributors of the GAP) or projects (that in some way inspired the
+GAP).
+
+An **unimplemented** GAP may also include the following sections. These are
+optional.
+
+1. **Prototype Implementation:** A description or link to a prototype of the
+proposed changes.
+
+When a GAP is implemented, the document must be updated to reflect this in the
+metadata section. Additionally, the following sections must be added to the
+document.
+
+1. **Implementation:** A description of how the GAP was implemented in gittuf.
+If a prototype implementation was accepted as the final implementation, this
+section may indicate as such and refer to the prototype implementation section.
+
+If the features proposed in an implemented GAP are later removed, the GAP must
+be updated to reflect this in the document's metadata section. The GAP's
+reasoning must also be updated to indicate why the feature was removed.
+
+## GAP Responsibilities
+
+The participants in a GAP have specific responsibilities.
+
+### Sponsor
+
+Every GAP must have at least one sponsor. There is no limit to the number of
+sponsors a GAP may have. Each sponsor must be listed in the GAP's metadata
+section. The sponsors are the original proposers of a GAP and take on the
+responsibility of authoring the document and submitting it for review by the
+gittuf maintainers. Additionally, the sponsors should update the GAP based on
+feedback or changes to the corresponding implementations, thus ensuring the
+document reflects the latest status of the proposed changes.
+
+### Contributor
+
+A GAP may have one or more contributors. These are members of the community who
+contribute to the GAP but do not wish to sponsor it. Each contributor must be
+listed in the GAP's metadata section.
+
+### gittuf Maintainers
+
+Ultimately, the gittuf maintainers are responsible for overseeing GAPs and
+keeping them updated. The responsibilities of the gittuf maintainers include
+(but are not limited to):
+
+* Engaging in discussions of problems to determine if a GAP is necessary
+* Reviewing and providing feedback on a GAP in a timely manner with a focus on
+  the impact of the proposed changes on gittuf's security model
+* Ensuring the GAP follows the prescribed format
+* Updating merged GAPs (if the sponsors do not) to reflect the state of their
+  implementation; for example, if a GAP is implemented, if an implemented GAP
+  feature is removed, or a GAP feature is updated significantly, the maintainers
+  must ensure this is reflected in the living GAP document
+
+Changes to GAP documents are subject to approval from the same threshold of
```

Comment:
Considering that a GAP can potentially have a bigger impact, are we comfortable with the same threshold required for a (potentially much more minor) implementation change?

Reply:
This is one of the processes I scaled back (earlier there were stricter review requirements for GAPs).

I think the same threshold is fine because ultimately, the big changes can be merged into the implementation without any reasoned documentation and discussion (what a GAP provides us) by a "small" threshold of reviews. So requiring GAPs at all is already an improvement I think. If we have an explosion in number of GAPs and feature additions, we can revise this at a later date.